frontend/packages/kubevirt-plugin/src/components/vms/vm-environment/selectors.ts

```diff
@@ -0,0 +1,131 @@
+import * as _ from 'lodash';
+import { sourcesEnum } from './types';
+import { getRandomChars } from '@console/dev-console/src/utils/shared-submit-utils';
+import { V1Volume } from '../../../types/vm/disk/V1Volume';
+
+export const getSourceName = (source): string => {
+  if (!source) {
+    return '';
+  }
+
+  return (
+    source.configMapRef?.name || source.secretRef?.name || source.serviceAccountRef?.name || ''
+  );
+};
+
+export const getSourceKind = (envDisk: (string | object | number)[]): string => {
+  // get only disk[1] as string
+  const ref = envDisk[1];
+  if (Object.keys(ref).includes('configMapRef')) {
+    return 'configMap';
+  }
+
+  if (Object.keys(ref).includes('secretRef')) {
+    return 'secret';
+  }
+
+  if (Object.keys(ref).includes('serviceAccountRef')) {
+    return 'serviceAccount';
+  }
+
+  return null;
+};
+
+export const getNewDiskName = (sourceName) => `${sourceName}-${getRandomChars(6)}-disk`;
+
+export const areEnvDisksEqual = (envDisk1, envDisk2): boolean => {
+  if (!envDisk1 && !envDisk2) {
+    return true;
+  }
+  if (!envDisk1 || !envDisk2) {
+    return false;
+  }
+
+  return envDisk1[0] === envDisk2[0] && _.isEqual(envDisk1[1], envDisk2[1]);
+};
+
+export const getEnvDiskRefKind = (envDisk) => {
+  return envDisk[1] && Object.keys(envDisk[1])[0];
+};
+
+export const toListObj = (kind: string, items) => {
+  return {
+    apiVersion: 'v1',
+    kind,
+    metadata: {},
+    items,
+  };
+};
+
+export const getAvailableSources = (allSources, usedSources) =>
+  allSources.filter(
+    (src) => !usedSources.find((ed) => ed[1][getEnvDiskRefKind(ed)].name === src.metadata.name),
+  );
+
+export const getNewEnvVarSource = (kind: string, name: string) => {
+  if (!kind || !name) {
+    return null;
+  }
+
+  switch (kind) {
+    case sourcesEnum.configMapKind:
+      return { configMapRef: { name } };
+    case sourcesEnum.secretKind:
+      return { secretRef: { name } };
+    case sourcesEnum.serviceAccountKind:
+      return { serviceAccountRef: { name } };
+    default:
+      return null;
+  }
+};
+
+export const getSerialNumber = () =>
+  getRandomChars(10)
```

Comment:
why not getRandomChars(16) ? what am I missing ?

Reply:
because the limit is 10.
if you give it 16 you will get 10 eventually.